src/igmpv3.h

```diff
@@ -21,6 +21,21 @@
 *   igmpv3.h - Header file for common IGMPv3 includes.
 */
 
+/*
+ * IGMP v3 query format.
+ */
+struct igmpv3 {
+	u_int8_t		igmp_type;	/* version & type of IGMP message  */
+	u_int8_t		igmp_code;	/* subtype for routing msgs        */
+	u_int16_t		igmp_cksum;	/* IP-style checksum               */
+	struct in_addr	igmp_group;	/* group address being reported    */
+					/*  (zero for queries)             */
+	u_int8_t		igmp_misc;	/* reserved/suppress/robustness    */
+	u_int8_t		igmp_qqi;	/* querier's query interval        */
+	u_int16_t		igmp_numsrc;	/* number of sources               */
+	/*struct in_addr	igmp_sources[1];*/ /* source addresses */
```

Comment:
Source addresses is part of _IGMPv3 Membership Query_ structure, so let it there. igmpproxy already depends on C99, so you can write `igmp_sources[];` -- it does not change `sizeof()` and therefore should be safe.

Reply:
See comment above, I have copied it from "netinet/igmp.h" of xBSD.

Reply:
But it does not mean that we should have _incomplete_ structure too.